### DISCUSSION TITLE
How can I create a new UI Form for Client Plugin in Netbeans?

### DISCUSSION THREAD

**Discussion by jbravo94** [nextgenhealthcare/connect#4546] [Q&A]
Dear Connect Community,

i want to create a custom Plugin since i need a custom FrameHandler for the TCP Server.

I can compile successfully the MLLPPlugin for testing and also install it but unfortunately i cannot make changes in the UI form in Netbeans.

Do i miss something when i get following errors:

![image](https://user-images.githubusercontent.com/26746017/118416814-b65dcf80-b6b1-11eb-830d-316d53677e00.png)

![image](https://user-images.githubusercontent.com/26746017/118416805-b067ee80-b6b1-11eb-9242-95b0269ca272.png)

I tried to add everything to the classpath:

![image](https://user-images.githubusercontent.com/26746017/118416833-cfff1700-b6b1-11eb-92b4-d2934a3f5214.png)

I also tried to add a different component as bean (`com.mirth.connect.client.ui.ByteAbbreviationList`):

![image](https://user-images.githubusercontent.com/26746017/118416936-426ff700-b6b2-11eb-94f1-ab61b203db41.png)

but only receive this error

`java.lang.NoClassDefFoundError: org/jdesktop/swingx/JXList`

Do i need to add the swingx library manually from here https://repo1.maven.org/maven2/org/swinglabs/swingx/swingx-all/ ?

Although `swingx-core-1.6.2.jar` exist already in the `client/lib` directory.

Is there maybe a new approach also for FrameHandler Plugins because i saw that OpenJavaFX library should be used somewhere else?

Currently i have installed latest Netbeans 12.0 with AdoptOpenJDK 10 (was currently best fit for compilation for me).

I kindly ask for your help.

Sincerely,
Johannes

**Comment by jbravo94** [comment]
@kpalang Maybe you stumbled also across this problem since you are currently developing your own plugin in #4510 and maybe have some suggestions for me? :)

**Reply by kpalang** [reply]
I haven't seen your source code but my first suspicion is that you haven't added Mirth jars to your project. You can take a look at a maven repo I'm hosting at https://nexus.kaurpalang.com/. The plugin you mentioned uses that repository.

**Reply by jbravo94** [reply]
Cool thanks!
I will have a look at it.

**Reply by kpalang** [reply]
If you want, you can also find me on Mirth Slack with the same username
